server/src/main/scala/ch.epfl.scala.index.server/Api.scala

```diff
@@ -0,0 +1,300 @@
+package ch.epfl.scala.index
+package server
+
+import model._
+import release.{SemanticVersion, ScalaTarget, MavenReference}
+import misc.Pagination
+
+import data.elastic._
+import data.cleanup.ArtifactNameParser
+
+import com.sksamuel.elastic4s._
+import ElasticDsl._
+import org.elasticsearch.search.sort.SortOrder
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.reflectiveCalls
+
+class Api(github: Github)(implicit val ec: ExecutionContext) {
```

Comment:
The name `Api` does not say much about the class itself. It's actually only doing search and kind of data access layer.
Also i'd split project index from release index to have it separated and more clear.

A package like "search", "dataaccess", "repository" would be useful.

Also method's could get a doc block comment to know what they are made for,